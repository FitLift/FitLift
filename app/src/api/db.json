{
  "currentExercises": [
    {
      "reps": 12,
      "time": 1517968200,
      "type": "Bicep Curls"
    },
    {
      "reps": 1,
      "time": 1517968200,
      "type": "Shoulder Press"
    },
    {
      "reps": 6,
      "time": 1517968200,
      "type": "Lateral Raises"
    }
  ]
}
