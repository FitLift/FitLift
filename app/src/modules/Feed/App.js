import React from 'react';
import PropTypes from 'prop-types';
import { Button, StyleSheet, Text, View } from 'react-native';

const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
    backgroundColor: '#F5FCFF',
    flex: 1,
    justifyContent: 'center',
  },
  welcome: {
    fontSize: 20,
    margin: 10,
    textAlign: 'center',
  },
});

const Friends = ({ navigation }) => (
  <View style={styles.container}>
    <Text style={styles.instructions}>
      Feed
    </Text>
    <Button
      onPress={() => navigation.dispatch({ type: 'Feed' })}
      title="Feed"
    />
  </View>
);

Friends.navigationOptions = {
  title: 'Feed',
};

export default Friends;
