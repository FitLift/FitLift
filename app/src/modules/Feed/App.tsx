import React from 'react';
import PropTypes from 'prop-types';
import { Button, StyleSheet, Text, View } from 'react-native';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#F5FCFF',
  },
  welcome: {
    fontSize: 20,
    textAlign: 'center',
    margin: 10,
  },
});

const Friends = ({ navigation }) => (
  <View style={styles.container}>
    <Text style={styles.instructions}>
      Feed
    </Text>
    <Button
      onPress={() => navigation.dispatch({ type: 'Feed' })}
      title="Feed"
    />
  </View>
);

Friends.propTypes = {
  navigation: PropTypes.object.isRequired,
};

Friends.navigationOptions = {
  title: 'Feed',
};

export default Friends;
