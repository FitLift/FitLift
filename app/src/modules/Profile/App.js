import React from 'react';
import PropTypes from 'prop-types';
import { Button, StyleSheet, Text, View } from 'react-native';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#F5FCFF',
  },
  welcome: {
    fontSize: 20,
    textAlign: 'center',
    margin: 10,
  },
});

const Profile = ({ navigation }) => (
  <View style={styles.container}>
    <Text style={styles.instructions}>
      Profile
    </Text>
    <Button
      onPress={() => navigation.dispatch({ type: 'Logout' })}
      title="Log Out"
    />
  </View>
);

Profile.propTypes = {
  navigation: PropTypes.object.isRequired,
};

Profile.navigationOptions = {
  title: 'Profile',
};

export default Profile;
