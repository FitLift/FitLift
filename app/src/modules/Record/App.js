import React from 'react';
import PropTypes from 'prop-types';
import { Button, StyleSheet, Text, View } from 'react-native';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#F5FCFF',
  },
  welcome: {
    fontSize: 20,
    textAlign: 'center',
    margin: 10,
  },
});

const record = ({ navigation }) => (
  <View style={styles.container}>
    <Text style={styles.instructions}>
      record
    </Text>
    <Button
      onPress={() => navigation.dispatch({ type: 'record' })}
      title="record"
    />
  </View>
);

record.propTypes = {
  navigation: PropTypes.object.isRequired,
};

record.navigationOptions = {
  title: 'record',
};

export default record;
