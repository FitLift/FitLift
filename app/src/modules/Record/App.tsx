import React from 'react';
import PropTypes from 'prop-types';
import { Button, StyleSheet, Text, View } from 'react-native';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#F5FCFF',
  },
  welcome: {
    fontSize: 20,
    textAlign: 'center',
    margin: 10,
  },
});

const Record = ({ navigation }) => (
  <View style={styles.container}>
    <Text style={styles.instructions}>
      Record
    </Text>
    <Button
      onPress={() => navigation.dispatch({ type: 'Logout' })}
      title="Logout"
    />
  </View>
);

Record.propTypes = {
  navigation: PropTypes.object.isRequired,
};

Record.navigationOptions = {
  title: 'Record',
};

export default Record;
