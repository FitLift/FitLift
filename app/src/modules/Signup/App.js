import React from 'react';
import PropTypes from 'prop-types';
import { Button, StyleSheet, Text, View } from 'react-native';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#F5FCFF',
  },
  welcome: {
    fontSize: 20,
    textAlign: 'center',
    margin: 10,
  },
});

const Login = ({ navigation }) => (
  <View style={styles.container}>
    <Text style={styles.welcome}>
      login
    </Text>
    <Button
      onPress={() => navigation.dispatch({ type: 'Signup' })}
      title="Signup"
    />
  </View>
);

Login.navigationOptions = {
  title: 'Sign up',
};

export default Login;
