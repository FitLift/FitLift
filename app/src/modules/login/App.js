import React from 'react';
import { Button, Text, View } from 'react-native';

const login = ({ navigation }) => (
  <View>
    <Text>login</Text>
    <Button
      onPress={() => navigation.dispatch({ type: 'login' })}
      title="Log in"
    />
  </View>
);

login.navigationOptions = {
  title: 'Log In'
};

export default login;
