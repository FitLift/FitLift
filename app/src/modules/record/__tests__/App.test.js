import React from 'react';
import renderer from 'react-test-renderer';
import { App, mapStateToProps } from '../App';
import { new_exercises as newExercises } from '../../../api/db.json';

it('renders without crashing', () => {
  const props = {
    navigation: {},
    newExercises,
    record: {
      isLoading: false,
    },
  };
  const rendered = renderer.create(
    <App
      fetchNewExercises={jest.fn()}
      {...props}
    />,
  ).toJSON();
  expect(rendered).toMatchSnapshot();
});

describe('mapStateToProps tests', () => {
  it('should work with intial state', () => {
    expect(mapStateToProps({
      db: {
        newExercises: [],
      },
      record: {
        isLoading: false,
      },
    })).toEqual({
      newExercises: [],
      record: {
        isLoading: false,
      },
    });
  });

  it('should work when loading', () => {
    expect(mapStateToProps({
      db: {
        newExercises: [],
      },
      record: {
        isLoading: true,
      },
    })).toEqual({
      newExercises: [],
      record: {
        isLoading: true,
      },
    });
  });

  it('should work with new exercises', () => {
    expect(mapStateToProps({
      db: {
        newExercises,
      },
      record: {
        isLoading: false,
      },
    })).toEqual({
      newExercises,
      record: {
        isLoading: false,
      },
    });
  });
});
