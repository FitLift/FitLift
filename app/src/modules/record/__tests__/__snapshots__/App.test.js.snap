// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F5FCFF",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Exercise: 
    Bicep Curls
     reps: 
    12
     
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Exercise: 
    Shoulder Press
     reps: 
    5
     
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Exercise: 
    Lateral Raises
     reps: 
    6
     
  </Text>
</View>
`;
